docs/coding-guidelines/libraries-packaging.md

```diff
@@ -0,0 +1,101 @@
+
+Removing a library from this transport package is a breaking change and should be avoided.
+
+## NuGet package
+
+Libraries to be packaged must be referenced by the [traversal packaging project](../../src/libraries/libraries-packages.proj) and set `IsPackable` to true.  By default, all `Libraries/*/src` projects are considered for packaging.
+
+Package versions and shipping state should be controlled using the properties defined by the [Arcade SDK](https://github.com/dotnet/arcade/blob/master/Documentation/ArcadeSdk.md#project-properties-defined-by-the-sdk).  Typically libraries should not need to explicitly set any of these properties.
+
+Most metadata for packages is controlled centrally in the repository and individual projects need not make any changes to these.  One property is required to be set in each project: `PackageDescription`.  This should be set to a descriptive summary of the purpose of the package, and a list of common entry-point types for the package: to aide in search engine optimization.  Example:
+```xml
+<PackageDescription>Logging abstractions for Microsoft.Extensions.Logging.
+
+Commonly Used Types:
+Microsoft.Extensions.Logging.ILogger
+Microsoft.Extensions.Logging.ILoggerFactory
+Microsoft.Extensions.Logging.ILogger&lt;TCategoryName&gt;
+Microsoft.Extensions.Logging.LogLevel
+Microsoft.Extensions.Logging.Logger&lt;T&gt;
+Microsoft.Extensions.Logging.LoggerMessage
+Microsoft.Extensions.Logging.Abstractions.NullLogger</PackageDescription>
+```
+
+Package content can be defined using any of the publicly defined Pack inputs: https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets
+
+### TargetFrameworks
+
+By default all TargetFrameworks listed in your project will be included in the package.  You may exclude sepecific TargetFrameworks by setting `ExcludeFromPackage` on that framework.
+```xml
+  <PropertyGroup>
+    <ExcludeFromPackage Condition="'$(TargetFramework)' == 'net5.0'">true</ExcludeFromPackage>
+  </PropertyGroup>
+```
+
+A common pattern is to build for the latest .NET version, for example to include a library in the shared framework or a transport package, but then excluded this from the NuGet package.  This can be done to avoid growing the NuGet package in size.  To do this set
+```xml
+  <PropertyGroup>
+    <ExcludeCurrentNetCoreAppFromPackage>true</ExcludeCurrentNetCoreAppFromPackage>
+  </PropertyGroup>
+```
+
+When excluding TargetFrameworks from a package special care should be taken to ensure that the builds included are equivalent to those excluded.  Avoid ifdef'ing the implementation only in an excluded TargetFramework.  Doing so will result in testing something different than what we ship, or shipping a nuget package that degrades the shared framework.
```

Comment:
```suggestion
When excluding TargetFrameworks from a package special care should be taken to ensure that the builds included are equivalent to those excluded. Avoid ifdef'ing the implementation only in an excluded TargetFramework. Doing so will result in testing something different than what we ship, or shipping a nuget package that degrades the shared framework.
```

Are all these double spaces after the `.` intentional?

Reply:
This writing style was what was hammered into me when I was in school and it’s just muscle memory.

I checked the current MLA style and I guess it’s still an open debate 😅
https://style.mla.org/formatting-papers/
> Leave one space after a period or other concluding punctuation mark, **unless your instructor prefers two spaces.**

Do we have a style guideline on dotnet/runtime to dictate this?

Reply:
I have no idea. It seems like markdown lint doesn't have a tule for it as it is passing, but I don't know if people have strong opinions.

Reply:
I guess 95% of our markdown files don't use double spaces but I don't have a strong opinion either. Consistency would be nice though.

Reply:
I'm fine to go with the flow of the repo here. It might take me some time to retrain my muscle memory. I had to correct both sentences in this very comment to remove the double space 😆

Reply:
I also had "two spaces after a period" hammered into me.  My fingers do it automatically.  I can't convince them otherwise.